tests/executors/test_celery_executor.py

```diff
@@ -248,6 +249,22 @@ def test_cleanup_stuck_queued_tasks(self, mock_fail):
         assert app.control.revoke.called_with("231")
         assert mock_fail.called_once()
 
+    @conf_vars({("celery", "result_backend_engine_options"): '{"pool_recycle": 1800}'})
+    @mock.patch("celery.Celery")
+    def test_result_backend_engine_options(self, mock_celery):
+        import importlib
+
+        from airflow.config_templates import default_celery
+
+        # relaod celery conf to apply the new config
+        importlib.reload(default_celery)
+        # reload celery_executor to recreate the celery app with new config
+        importlib.reload(celery_executor)
```

Comment:
This will create a logical conflict (git won't be able to detect) with #31001
Whoever merges first will have to resolve that, but it is a very easy fix (just reload the utils module instead)

Reply:
My change has been merged, so you'll have to resolve this conflict.

The celery app now lives in the celery_executor_utils module, so that is what needs to be reloaded to get a new app (note, the app is still available as an attribute on the celery_executor module thanks to PEP562, but it does not live there).

Reply:
I just updated it